routers/user/home.go

```diff
@@ -214,48 +214,28 @@ func Issues(ctx *context.Context) {
 
 	// Get repositories.
 	var err error
-	var repos []*models.Repository
-	userRepoIDs := make([]int64, 0, len(repos))
+	var userRepoIDs []int64
 	if ctxUser.IsOrganization() {
 		env, err := ctxUser.AccessibleReposEnv(ctx.User.ID)
 		if err != nil {
 			ctx.Handle(500, "AccessibleReposEnv", err)
 			return
 		}
-		repos, err = env.Repos(1, ctxUser.NumRepos)
+		userRepoIDs, err = env.RepoIDs(1, ctxUser.NumRepos)
 		if err != nil {
-			ctx.Handle(500, "GetRepositories", err)
+			ctx.Handle(500, "env.RepoIDs", err)
 			return
 		}
-
-		for _, repo := range repos {
-			if (isPullList && repo.NumPulls == 0) ||
-				(!isPullList &&
-					(!repo.EnableUnit(models.UnitTypeIssues) || repo.NumIssues == 0)) {
-				continue
-			}
-
-			userRepoIDs = append(userRepoIDs, repo.ID)
-		}
-
-		if len(userRepoIDs) <= 0 {
-			userRepoIDs = []int64{-1}
-		}
-
 	} else {
-		if err := ctxUser.GetRepositories(1, ctx.User.NumRepos); err != nil {
-			ctx.Handle(500, "GetRepositories", err)
+		userRepoIDs, err = ctxUser.GetAccessRepoIDs()
+		if err != nil {
+			ctx.Handle(500, "ctxUser.GetAccessRepoIDs", err)
 			return
 		}
-		repos = ctxUser.Repos
+	}
 
-		for _, repo := range repos {
-			if (isPullList && repo.NumPulls == 0) ||
-				(!isPullList &&
-					(!repo.EnableUnit(models.UnitTypeIssues) || repo.NumIssues == 0)) {
-				continue
-			}
-		}
+	if len(userRepoIDs) <= 0 {
+		userRepoIDs = []int64{-1}
```

Comment:
I believe this check isn't necessary; both `Issues(..)` and `GetUserIssueStats(..)` check for empty slices.